src/test/tst_value.cpp

```diff
@@ -89,44 +70,148 @@ void TestValue::tst_compare()
     QVERIFY_EXCEPTION_THROWN(v1 <= v2, std::invalid_argument);
     QVERIFY_EXCEPTION_THROWN(v1 > v2, std::invalid_argument);
     QVERIFY_EXCEPTION_THROWN(v1 < v2, std::invalid_argument);
+}
+
+void TestValue::tst_valueBool(){
+    Value v;
+
+    //VALUE TESTS
+    //case where v is set to true
+    v = Value(true);
+
+    //Testing validity
+    QVERIFY(v.isValid() == true);
 
-    // compare bool
+    //Testing type
+    QVERIFY(v.type() == Value::BOOL);
+
+    //Testing value
+    QVERIFY(v.toBool() == true);
+    QCOMPARE(v.toQString(), QString("1"));
+
+    //case where v is set to false
+    v = Value(false);
+
+    //Testing validity
+    QVERIFY(v.isValid() == true);
+
+    //Testing type
+    QVERIFY(v.type() == Value::BOOL);
+
+    //Testing value
+    QVERIFY(v.toBool() == false);
+
+    //COMPARISON TESTS
+    Value v1, v2;
     v1 = Value(true), v2 = Value(false);
     QCOMPARE(v1 == v2, false);
     QCOMPARE(v1 != v2, true);
     QCOMPARE(v1 >= v2, true);
     QCOMPARE(v1 <= v2, false);
     QCOMPARE(v1 >  v2, true);
     QCOMPARE(v1 <  v2, false);
+}
 
-    // compare char
-    v1 = Value('d'), v2 = Value('c');
+void TestValue::tst_valueDouble(){
+    Value v;
+
+    //VALUE TESTS
+    v = Value(4.5123788);
+
+    //Testing validity
+    QVERIFY(v.isValid() == true);
+
+    //Testing type
+    QVERIFY(v.type() == Value::DOUBLE);
+
+    //Testing value
+    QCOMPARE(v.toDouble(), 4.5123788);
+    QCOMPARE(v.toQString(), QString("4.5123788"));
+
+    //COMPARISON TESTS
+    Value v1, v2;
+    v1 = Value(100.2), v2 = Value(100.1);
     QCOMPARE(v1 == v2, false);
     QCOMPARE(v1 != v2, true);
     QCOMPARE(v1 >= v2, true);
     QCOMPARE(v1 <= v2, false);
     QCOMPARE(v1 >  v2, true);
     QCOMPARE(v1 <  v2, false);
+}
 
-    // compare double
-    v1 = Value(100.2), v2 = Value(100.1);
+void TestValue::tst_valueInt(){
+    Value v;
+
+    //VALUE TESTS
+    v = Value(-10);
+
+    //Testing validity
+    QVERIFY(v.isValid() == true);
+
+    //Testing type
+    QVERIFY(v.type() == Value::INT);
+
+    //Testing value
+    QCOMPARE(v.toInt(), -10);
+    QCOMPARE(v.toQString(), QString("-10"));
+
+    //COMPARISON TESTS
+    Value v1, v2;
+    v1 = Value(100), v2 = Value(50);
     QCOMPARE(v1 == v2, false);
     QCOMPARE(v1 != v2, true);
     QCOMPARE(v1 >= v2, true);
     QCOMPARE(v1 <= v2, false);
     QCOMPARE(v1 >  v2, true);
     QCOMPARE(v1 <  v2, false);
+}
+void TestValue::tst_valueChar(){
+    Value v;
 
-    // compare int
-    v1 = Value(100), v2 = Value(50);
+    //VALUE TESTS
+    const char* chr = "abc£ãã&";
```

Comment:
a `const char*` is actually a string
you should be testing a simple `char` here, for example:
``` c
char chr = 'a';
```